src/badge/minter.cairo

```diff
@@ -0,0 +1,231 @@
+    // Implicit args:
+    //   syscall_ptr(felt*)
+    //   pedersen_ptr(HashBuiltin*)
+    //   range_check_ptr
+    // Explicit args:
+    //   new_signer_public_key(felt): The new public key
+    // Returns:
+    //   None
+    alloc_locals;
+    Ownable.assert_only_owner();
+    signer_public_key.write(new_signer_public_key);
+    return ();
+}
+
+@external
+func setBadgeContractAddress{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+    new_badge_contract_address: felt
+) -> () {
+    // Desc:
+    //   Set the address of the badge contract
+    // Implicit args:
+    //   syscall_ptr(felt*)
+    //   pedersen_ptr(HashBuiltin*)
+    //   range_check_ptr
+    // Explicit args:
+    //   new_badge_contract_address(felt): The new badge contract address
+    // Returns:
+    //   None
+    alloc_locals;
+    Ownable.assert_only_owner();
+    badge_contract_address.write(new_badge_contract_address);
+    return ();
+}
+
+@external
+func transferOwnership{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+    newOwner: felt
+) {
+    // Desc:
+    //   Transfer ownership to a new owner
+    // Implicit args:
+    //   syscall_ptr(felt*)
+    //   pedersen_ptr(HashBuiltin*)
+    //   range_check_ptr
+    // Explicit args:
+    //   newOwner(felt): Address of the new owner
+    // Returns:
+    //   None
+    Ownable.transfer_ownership(newOwner);
+    return ();
+}
+
+@external
+func renounceOwnership{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() {
+    // Desc:
+    //   Renounce ownership
+    // Implicit args:
+    //   syscall_ptr(felt*)
+    //   pedersen_ptr(HashBuiltin*)
+    //   range_check_ptr
+    // Returns:
+    //   None
+    Ownable.renounce_ownership();
+    return ();
+}
+
+@external
+func transferBadgeContractOwnership{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+    newOwner: felt
+) {
+    // Desc:
+    //   Transfer ownership of the badge contract to a new owner
+    // Implicit args:
+    //   syscall_ptr(felt*)
+    //   pedersen_ptr(HashBuiltin*)
+    //   range_check_ptr
+    // Explicit args:
+    //   newOwner(felt): Address of the new owner
+    // Returns:
+    //   None
+    Ownable.assert_only_owner();
+    let (contract_address) = badge_contract_address.read();
+    IBadgeContract.transferOwnership(contract_address, newOwner);
+    return ();
+}
+
+//
+// UPGRADABILITY
+//
+
+@external
+func upgrade{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+    new_implementation: felt
+) {
+    // Desc:
+    //   Upgrade the implementation of the proxy
+    // Implicit args:
+    //   syscall_ptr(felt*)
+    //   pedersen_ptr(HashBuiltin*)
+    //   range_check_ptr
+    // Explicit args:
+    //   new_implementation(felt): Address of the new implementation
+    // Returns:
+    //   None
+    Ownable.assert_only_owner();
+    Proxy._set_implementation_hash(new_implementation);
+    return ();
+}
```

Comment:
I think you can add all other upgradable functions such as, setAdmin, getAdmin, getImplementationHash